packages/governance/src/governor/extensions/governor_votes_quorum_fraction.cairo

```diff
@@ -0,0 +1,212 @@
+    > of IQuorumFraction<ComponentState<TContractState>> {
+        /// Returns the token that voting power is sourced from.
+        fn token(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.Governor_token.read()
+        }
+
+        /// Returns the current quorum numerator.
+        fn current_quorum_numerator(self: @ComponentState<TContractState>) -> u256 {
+            self.Governor_quorum_numerator_history.deref().latest()
+        }
+
+        /// Returns the quorum numerator at a specific timepoint.
+        fn quorum_numerator(self: @ComponentState<TContractState>, timepoint: u64) -> u256 {
+            // Optimistic search: check the latest checkpoint.
+            // The initializer call ensures that there is at least one checkpoint in the history.
+            //
+            // NOTE: This optimization is specially helpful when the supply is not updated often.
+            let (_, key, value) = self
+                .Governor_quorum_numerator_history
+                .deref()
+                .latest_checkpoint();
+
+            if key <= timepoint {
+                return value;
+            }
+
+            // Fallback to the binary search
+            self.Governor_quorum_numerator_history.deref().upper_lookup(timepoint)
+        }
+
+        /// Returns the quorum denominator.
+        fn quorum_denominator(self: @ComponentState<TContractState>) -> u256 {
+            100
+        }
+    }
+
+    //
+    // Internal
+    //
+
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        +GovernorComponent::GovernorVotesTrait<TContractState>,
+        impl Governor: GovernorComponent::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Initializes the component by setting the votes token and the
+        /// initial quorum numerator value.
+        ///
+        /// Requirements:
+        ///
+        /// - `votes_token` must not be zero.
+        /// - `quorum_numerator` must be less than `quorum_denominator`.
+        ///
+        /// Emits a `QuorumNumeratorUpdated` event.
+        fn initializer(
+            ref self: ComponentState<TContractState>,
+            votes_token: ContractAddress,
+            quorum_numerator: u256
+        ) {
+            assert(votes_token.is_non_zero(), Errors::INVALID_TOKEN);
+
+            self.Governor_token.write(votes_token);
+            self.update_quorum_numerator(quorum_numerator);
+        }
+
+        /// Updates the quorum numerator.
+        ///
+        /// Requirements:
+        ///
+        /// - `new_quorum_numerator` must be less than `quorum_denominator`.
+        ///
+        /// Emits a `QuorumNumeratorUpdated` event.
+        fn update_quorum_numerator(
+            ref self: ComponentState<TContractState>, new_quorum_numerator: u256
+        ) {
```

Comment:
Should we expose this so it can be updated (through governance)?

Reply:
Not sure if allow to update the quorum after the initialization is a good idea, for the sake of simplicity, we can leave the decision to the users for now, as they can expose the utility using assert_only_governance.

Reply:
Simplicity sounds good for now